### DISCUSSION TITLE
x-teleport with class selector

### DISCUSSION THREAD

**Discussion by Chandler-Zhu** [alpinejs/alpine#2940] [1. Help]
I tried to use `x-teleport` with `.class` selector but things didn't work as I expected
demo: https://jsfiddle.net/qdh1sLew/
When I click the toggle button only the first div with the `.feature` class has a circle append to it instead of all four of them

**Comment by GamesmenJordan** [comment]
This is intended functionality.

Per the documentation:

> The x-teleport selector can be any string you would normally pass into something like document.querySelector. **It will find the first element that matches**, be it a tag name (body), class name (.my-class), ID (#my-id), or any other valid CSS selector.

**Reply by Chandler-Zhu** [reply]
Maybe they could give `x-teleport` a `.all` modifier to make it more flexible
